tests/next_tests/ffront_tests/test_execution.py

```diff
@@ -333,6 +332,40 @@ def astype_fieldop_float(b: Field[[IDim], int64]) -> Field[[IDim], int32]:
     assert np.allclose(c_int32.array(), out_int_32)
 
 
+def test_dusk_indexfield(fieldview_backend):
+    a_I_arr = np.random.randn(size, size).astype("float64")
+    a_I_float = np_as_located_field(IDim, KDim)(a_I_arr)
+    offset_field_arr = np.asarray(np.ones((size, size)), dtype=int64)
+    offset_i_arr = np.asarray(np.ones((size,)), dtype=int64)
+    offset_field = np_as_located_field(IDim, KDim)(offset_field_arr)
+    offset_i = np_as_located_field(IDim)(offset_i_arr)
+    out_I_float = np_as_located_field(IDim, KDim)(np.zeros((size, size), dtype=float64))
+    out_I_float_1 = np_as_located_field(IDim, KDim)(np.zeros((size, size), dtype=float64))
+
+    @field_operator(backend=fieldview_backend)
+    def dusk_index_fo(
+        a: Field[[IDim, KDim], float64],
+        offset_field: Field[[IDim, KDim], int64],
+        offset_i: Field[[IDim], int64],
+    ) -> Field[[IDim, KDim], float64]:
+        a_0 = a(as_offset(IDim, offset_i))
+        a_i = a_0(as_offset(IDim, offset_field))
+        a_i_k = a_i(as_offset(KDim, offset_field))
```

Comment:
I only realized now that I was not precise in the description, it should be offsets not dimensions here, sorry for that (it's just that these offsets are then mapped to dimensions in the offset provider, that's why I made this mistake in the description)
```suggestion
        a_0 = a(as_offset(IOff, offset_i))
        a_i = a_0(as_offset(IOff, offset_field))
        a_i_k = a_i(as_offset(KOff, offset_field))
```